internal/persistence/postgres/postgres.go

```diff
@@ -0,0 +1,99 @@
+// Copyright © 2023 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgres
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"database/sql"
+
+	sq "github.com/Masterminds/squirrel"
+	migratedb "github.com/golang-migrate/migrate/v4/database"
+	"github.com/golang-migrate/migrate/v4/database/postgres"
+	"github.com/hyperledger/firefly-common/pkg/config"
+	"github.com/hyperledger/firefly-common/pkg/dbsql"
+	"github.com/hyperledger/firefly-transaction-manager/internal/persistence"
+
+	// Import pq driver
+	_ "github.com/lib/pq"
+)
+
+// Postgres provider implementation
+type Postgres struct {
+	dbsql.Database
+}
+
+var psql *Postgres
+
+func NewPostgresPersistence(bgCtx context.Context, conf config.Section, nonceStateTimeout time.Duration) (persistence.Persistence, error) {
+	if err := psql.Database.Init(bgCtx, psql, conf); err != nil {
+		return nil, err
+	}
+	return newSQLPersistence(bgCtx, &psql.Database, conf, nonceStateTimeout)
+}
+
+func (psql *Postgres) Name() string {
+	return "postgres"
+}
+
+func (psql *Postgres) SequenceColumn() string {
+	return "seq"
+}
+
+func (psql *Postgres) MigrationsDir() string {
+	return psql.Name()
+}
+
+// Attempt to create a unique 64-bit int from the given name, by selecting 4 bytes from the
+// beginning and end of the string.
+func lockIndex(lockName string) int64 {
+	if len(lockName) >= 4 {
+		lockName = lockName[0:4] + lockName[len(lockName)-4:]
+	}
+	return big.NewInt(0).SetBytes([]byte(lockName)).Int64()
+}
+
+func (psql *Postgres) Features() dbsql.SQLFeatures {
+	features := dbsql.DefaultSQLProviderFeatures()
+	features.PlaceholderFormat = sq.Dollar
+	features.UseILIKE = false // slower than lower()
+	features.AcquireLock = func(lockName string) string {
```

Comment:
Doesn't look like this functionality is ever used, so it's questionable if we should require implementing it...